main.js

```diff
@@ -1,158 +1,175 @@
-import { app, BrowserWindow, screen, Tray, Menu, nativeImage, ipcMain } from "electron";
+import { app, BrowserWindow, screen, Tray, Menu, nativeImage, ipcMain, dialog } from "electron";
 import { fileURLToPath } from "url";
 import path, { dirname } from "path";
 import dotenv from "dotenv";
+import fs from "fs";
 
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = dirname(fileURLToPath(import.meta.url));
 
 
 let welcomeView; // variable for the welcome page
 let mainView; // variable for the main text edit page
-
+let tray = null; // Tray should be initialized properly
 
 // load env variables
 dotenv.config();
 
-
 const renderMainWindow = () => {
-
-  // Get the primary display (the main screen the app will open on)
   const { width, height } = screen.getPrimaryDisplay().workAreaSize;
 
-  // Set width and height as percentages of the screen's size
   const windowWidth = Math.floor(width * 0.8); // 80% of screen width
   const windowHeight = Math.floor(height * 0.8); // 80% of screen height
 
   mainView = new BrowserWindow({
     width: windowWidth,
     height: windowHeight,
     webPreferences: {
-      preload: path.join(__dirname, 'preload.js'), // Use the new __dirname
+      preload: path.join(__dirname, "preload.js"),
       nodeIntegration: false,
       contextIsolation: true,
       zoomFactor: 1
     },
-    resizable: true, // Ensures the window can be resized
-    fullscreenable: true, // Allow fullscreen
+    resizable: true,
+    fullscreenable: true,
   });
-};
 
-  // Only open DevTools in local environment!
-  if (process.env.APP_ENV === 'local') {
+  if (process.env.APP_ENV === "local") {
```

Comment:
Why did you decide use Double quotes?

Reply:
When I incorrectly returned the function when creating the tray icon, you fixed my mistakes, including single commas in some places, so I thought "yes, indeed, this will look more consistent in the code." Therefore, I started using them whenever possible